hftbacktest/src/backtest/feemodel.rs

```diff
@@ -0,0 +1,54 @@
+use crate::types::Order;
+
+pub struct FeeArgs<'a> {
+    pub order: &'a Order,
+    pub amount: f64,
+}
+
+pub trait FeeModel {
+    // Calculates the fee amount.
+    fn amount(&self, args: FeeArgs) -> f64;
+}
+
+// fee based on a percentage of the order value, with the rate
+// depending on whether the order is a maker or taker.
+pub struct TradingValueFeeModel {
+    pub maker_fee: f64,
+    pub taker_fee: f64,
+}
+impl FeeModel for TradingValueFeeModel {
+    fn amount(&self, args: FeeArgs) -> f64 {
+        if args.order.maker {
+            self.maker_fee * args.amount
+        } else {
+            self.taker_fee * args.amount
+        }
+    }
+}
+// fee per trading quantity
+pub struct TradingQtyFeeModel {
+    pub per_qty_fee: f64,
+}
+impl FeeModel for TradingQtyFeeModel {
+    fn amount(&self, args: FeeArgs) -> f64 {
+        self.per_qty_fee * args.order.qty
```

Comment:
`order.qty` should be `order.exec_qty` to reflect the actual executed quantity especially in the case that the order is partially filled.

Reply:
Noted.